cmd/cli/plugin/secret/registry_secret_add.go

```diff
@@ -74,16 +87,41 @@ func registrySecretAdd(cmd *cobra.Command, args []string) error {
 			}
 		}
 		log.Info("\n")
-	}
-
-	pkgClient, err := tkgpackageclient.NewTKGPackageClient(registrySecretOp.KubeConfig)
-	if err != nil {
-		return err
-	}
+	} else {
+		secretExportList, err := pkgClient.ListSecretExports(registrySecretOp)
+		if err != nil {
+			return err
+		}
 
-	kc, err := kappclient.NewKappClient(registrySecretOp.KubeConfig)
-	if err != nil {
-		return err
+		for j := range secretExportList.Items {
+			exported = "not exported"
+			secretExport := secretExportList.Items[j]
+			if secretExport.Name == registrySecretOp.SecretName {
+
+				ns := &corev1.NamespaceList{}
```

Comment:
@vijaykatam I am bit confused about the namespace check ( for kubectl get ns), whether it should be a List or Get.
Is this on the same lines you had intended?

Reply:
Probably a code comment explaining this. Something like "Creating a SecretExport resource _X_ that conflicts with a previously defined SecretExport _Y_ that was exported to all namespaces could result in privilege escalation if the user does not access to other namespaces. This check prevents it by trying to list the namespaces"